packages/styled-base/types/index.d.ts

```diff
@@ -72,7 +71,7 @@ export interface CreateStyledComponentBase<
       ReactClassPropKeys
     > = Omit<InnerProps & ExtraProps, ReactClassPropKeys>
   >(
-    template: TemplateStringsArray,
+    template: TemplateStringsArray | CSSObject,
```

Comment:
This case is covered by previous overloading, so you don't need to add one.
`Interpolation<WithTheme<StyleProps, Theme>>` is a (structural) supertype of `CSSObject`.